spec/controllers/rents_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+describe RentsController do
+  let!(:starting_date) { Date.new 2017, 1, 1 }
+  let!(:ending_date) { Date.new 2018, 1, 1 }
+  let!(:book) { create :book }
+  let!(:book2) { create :book }
+  describe 'GET #index' do
+    context 'when an user is authenticated' do
+      include_context 'With an authenticated User'
+      let!(:rent1) do
+        Rent.create book: book, user: current_user,
+                    starting_date: starting_date, ending_date: ending_date
+      end
+      let!(:rent2) do
+        Rent.create book: book2, user: current_user,
+                    starting_date: starting_date, ending_date: ending_date
+      end
+      context 'when fetching all the rents' do
+        subject!(:http_response) { get :index, params: { user_id: current_user.id } }
```

Comment:
We know who is current_user because we have that information in request headers. Check when you are did something like this in shared_context 'With an authenticated User'
```rb
request.headers.merge! current_user.create_new_auth_token
```

Reply:
So, we don't need to find who is current_user using params :ok_man: